From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Tue, 13 Jul 2021 16:58:01 -0400
Subject: [PATCH] cmake: Make libuhd linking to Python library PRIVATE.

Linking with the Python library is only needed to embed the Python
interpreter when ENABLE_SIM=ON, so it's not a link that is needed by
anything linking with libuhd. When it is not PRIVATE, however, libpyuhd
links against the Python library too, which re-introduces the issues
fixed by #309.
---
 host/lib/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/host/lib/CMakeLists.txt b/host/lib/CMakeLists.txt
index 62e052d64..f5b9f1609 100644
--- a/host/lib/CMakeLists.txt
+++ b/host/lib/CMakeLists.txt
@@ -181,7 +181,7 @@ if(ENABLE_MPMD)
     list(APPEND libuhd_sources $<TARGET_OBJECTS:uhd_rpclib>)
 endif()
 add_library(uhd SHARED ${libuhd_sources})
-target_link_libraries(uhd ${Boost_LIBRARIES} ${libuhd_libs})
+target_link_libraries(uhd PUBLIC ${Boost_LIBRARIES} ${libuhd_libs})
 if(APPLE)
     target_link_options(uhd PRIVATE "LINKER:-undefined,dynamic_lookup")
     target_link_options(uhd PRIVATE "-flat_namespace")
@@ -213,7 +213,7 @@ if(ENABLE_SIM)
 
     # For PYUHD we don't link against the python libraries, but when calling
     # python instead of being called by it, we have to.
-    target_link_libraries(uhd ${PYTHON_LIBRARIES})
+    target_link_libraries(uhd PRIVATE ${PYTHON_LIBRARIES})
 
 endif(ENABLE_SIM)
 
-- 
2.46.0

